testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/ejb/client/exception/ExceptionClientHangsTestCase.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.as.test.manualmode.ejb.client.exception;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import org.jboss.arquillian.container.test.api.ContainerController;
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.container.test.api.TargetsContainer;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.logging.Logger;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Properties;
+
+/**
+ * Test for JBEAP-17612
+ *
+ * When exception is returned from server, WildFlyResponseClient hangs and client never receives the response.
+ * Test uses two deployments. One with the EJB client and one with the remote bean throwing exception.
+ * Test calls the client deployment and the client invokes the remote bean. Test should not keep hanging.
+ *
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class ExceptionClientHangsTestCase {
+
+    private static final Logger log = Logger.getLogger(ExceptionClientHangsTestCase.class);
+
+    private static final String CONTAINER = "jbossas-disable-assertions";
```

Comment:
@dcihak Can you comment here why this test needs a server with assertions disabled? It's not obvious so that may help future maintainers to understand. This seems like a good spot to explain as someone looking at what the arq container is for will likely end up here.

Reply:
@bstansberry  Thanks for the review. Comment was added.

Reply:
Thanks!